server/src/main/java/org/elasticsearch/cluster/coordination/JoinTaskExecutor.java

```diff
@@ -143,19 +148,43 @@ public JoinTaskExecutor(AllocationService allocationService, Logger logger, Rero
                     nodesChanged = true;
                     minClusterNodeVersion = Version.min(minClusterNodeVersion, node.getVersion());
                     maxClusterNodeVersion = Version.max(maxClusterNodeVersion, node.getVersion());
+                    // TODO do we need to check node.isMasterNode here for eligibility? I think since in the checks later master
+                    // eligibility will be taken into account anyway, here we may not need the check?
```

Comment:
Yes I think we should ignore master-ineligible nodes here.

Reply:
Done.